examples/example-app-router/src/i18n/request.ts

```diff
@@ -1,6 +1,29 @@
 import {notFound} from 'next/navigation';
 import {getRequestConfig} from 'next-intl/server';
 import {routing} from './routing';
+import { Formats } from 'next-intl';
+
+export const formats = {
+  dateTime: {
+    short: {
+      day: 'numeric',
+      month: 'short',
+      year: 'numeric'
+    }
+  },
+  number: {
+    precise: {
+      maximumFractionDigits: 5
+    }
+  },
+  list: {
+    enumeration: {
+      style: 'long',
+      type: 'conjunction'
+    }
+  }
+} as const satisfies Partial<Formats>
+
```

Comment:
Regarding tests, I currently use `example-app-router-playground` for type tests regarding `t`. Having the types directly inside `use-intl` and `next-intl` would definitely be preferable, but I haven't figured out a way to achieve that yet.

You can have a look here for inspiration:

https://github.com/amannn/next-intl/blob/2b363635db6542d726e6a4231bf1ac579fad7322/examples/example-app-router-playground/src/components/AsyncComponent.tsx#L15-L23

I'm also wondering if it'd be possible to test both cases, with an explicit declaration and without one, both in `example-app-router-playground` ([potential solution](https://www.perplexity.ai/search/in-typescript-override-a-globa-4qthXmTpRJengYPNaW9riA#0)).

- [ ] Tests for `useFormatter` and `getFormatter`

Reply:
will do!

Reply:
I was not able to adapt `example-app-router-playground` to have just one file with a different `IntlFormat` type. I guess we can do this same test but on a different app/workspace, where we don't have yet declared `IntlFormat` ?

Reply:
Oh, just to let you know, I can't click on these checkboxes you are adding :upside_down_face:

Reply:
I've added a few tests directly in `use-intl`, there we we have the use case of no strict formats available!